scripts/SREP/cluster-operator-status/script.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+# This script provides a rich summary of information concerning the status of 
+# degraded/unavailable OpenShift cluster operators.
+#
+# The script can specifically target a cluster operator through
+# the use of a "CLUSTER_OPERATOR" parameter
+
+set -e
+set -o nounset
+set -o pipefail
+
+
+# print_header prints a formatted section break header
+print_header() {
+    clusteroperator=$1
+    heading=$2
+    echo -e "--- [ CLUSTER OPERATOR '${clusteroperator}' ${heading} ] ---\n"
+}
+
+# get_summary prints the basic status for the given cluster operator parameter
+get_summary() {
+    clusteroperator=$1
+
+    # shellcheck disable=SC2016
+    available=$(jq -r --arg CLUSTER_OPERATOR "${clusteroperator}" '.items[] | select(.metadata.name == $CLUSTER_OPERATOR) | .status.conditions[] | select(.type == "Available") | .status' <<< "$OPERATOR_REPORT")
+    # shellcheck disable=SC2016
+    progressing=$(jq -r --arg CLUSTER_OPERATOR "${clusteroperator}" '.items[] | select(.metadata.name == $CLUSTER_OPERATOR) | .status.conditions[] | select(.type == "Progressing") | .status' <<< "$OPERATOR_REPORT")
+    # shellcheck disable=SC2016
+    degraded=$(jq -r --arg CLUSTER_OPERATOR "${clusteroperator}" '.items[] | select(.metadata.name == $CLUSTER_OPERATOR) | .status.conditions[] | select(.type == "Degraded") | .status' <<< "$OPERATOR_REPORT")
+
+    print_header "${clusteroperator}" "SUMMARY"
+    echo " - AVAILABLE:   ${available}"
+    echo " - DEGRADED:    ${degraded}"
+    echo " - PROGRESSING: ${progressing}"
+    echo -e "\n"
+}
+
+# get_description prints a detailed status description for the given cluster 
+# operator parameter
+get_description() {
+    clusteroperator=$1
+
+    if ! co_desc_report=$(oc describe clusteroperator "${clusteroperator}"); then
```

Comment:
```suggestion
    if ! co_desc_report=$(oc get co "${clusteroperator}" -ojson | jq -r '.status.conditions[] | select ((.type == "Degraded" or .type == "Unavailable") and (.status == "True")) | [.type, .message]'); then
```

Though the `oc describe` of the resource would have more details, I wonder if we can directly highlight the event message which the CO would update in the `.status.conditions` list. Else, another option can be to keep the `oc describe` in place as it is and have another section with "hints" of what the error could be? WDYT?

Reply:
I would add the `oc get co` after the report is printed (line 48)

Reply:
This makes sense, what I ended up doing was add it to the Summary section at the start to print out the `message` associated with each condition. It now behaves like this:

```
--- [ CLUSTER OPERATOR 'authentication' SUMMARY ] ---                                                                                                                                                                               
                                                                                                                                                                                                                                    
 - AVAILABLE:   True --> All is well
 - DEGRADED:    True --> APIServerDeploymentDegraded: 1 of 3 requested instances are unavailable for apiserver.openshift-oauth-apiserver ()
OAuthServerDeploymentDegraded: 1 of 3 requested instances are unavailable for oauth-openshift.openshift-authentication ()
 - PROGRESSING: False --> AuthenticatorCertKeyProgressing: All is well
```

Reply:
Thanks a lot! Yeah, this works too. Basically now the summary would give a glimpse of the issue at the beginning itself. If the user wants to know details then can refer to the logs etc.